packages/playwright-core/src/server/page.ts

```diff
@@ -227,6 +227,32 @@ export class Page extends SdkObject {
     this._browserContext.emit(event, ...args);
   }
 
+  async resetForReuse(metadata: CallMetadata) {
+    this.setDefaultNavigationTimeout(undefined);
+    this.setDefaultTimeout(undefined);
+
+    await this.removeExposedBindings();
+    await this.removeInitScripts();
+
+    // TODO: handle pending routes.
+    await this._frameManager.closeOpenDialogs();
+    await this.setClientRequestInterceptor(undefined);
```

Comment:
I think we should make `pageDispatcher._dispose` do this, otherwise it's going to leak and get notified about new routes.

Reply:
Done!